pkg/api/pod/warnings.go

```diff
@@ -303,6 +303,27 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 		return true
 	})
 
+	// Accumulate port names of all containers
+	allPortsName := map[string][]portBlock{}
+	pods.VisitContainersWithPath(podSpec, fieldPath.Child("spec"), func(c *api.Container, fldPath *field.Path) bool {
+		for i, port := range c.Ports {
+			if port.Name != "" {
+				k := port.Name
+				if others, found := allPortsName[k]; found {
+					for _, other := range others {
+						if port.Name == other.port.Name {
+							warnings = append(warnings, fmt.Sprintf("%s: duplicate port name definition with %s", fldPath.Child("ports").Index(i), other.field))
```

Comment:
include the name in the message

```suggestion
							warnings = append(warnings, fmt.Sprintf("%s: duplicate port name %q with %s", fldPath.Child("ports").Index(i), port.Name, other.field))
```